include/cuco/detail/bloom_filter/arrow_bf_policy_impl.cuh

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (c) 2024, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <cuco/detail/error.hpp>
+#include <cuco/hash_functions.cuh>
+
+#include <cuda/std/bit>
+#include <cuda/std/limits>
+#include <cuda/std/tuple>
+#include <cuda/std/type_traits>
+
+#include <cstdint>
+#include <nv/target>
+
+namespace cuco::detail {
+
+template <class T>
+class arrow_bf_policy_impl {
+ public:
+  using hasher    = cuco::xxhash_64<T>;  ///< xxhash_64 hasher for Arrow bloom filter policy
+  using word_type = std::uint32_t;       ///< uint32_t for Arrow bloom filter policy
+  using hash_argument_type = typename hasher::argument_type;
+  using hash_result_type   = decltype(std::declval<hasher>()(std::declval<hash_argument_type>()));
+
+  static constexpr uint32_t bits_set_per_block = 8;  ///< hardcoded  bits set per Arrow bf block
+  static constexpr uint32_t words_per_block    = 8;  ///< hardcoded words per Arrow bf block
+
+  __host__ __device__ explicit constexpr arrow_bf_policy_impl(std::uint32_t num_blocks, hasher hash)
```

Comment:
We are only using the `num_blocks` to validate the filter size to be in range: [32B, 128MB] (Arrow hardcoding). The constructor doesn't need `pattern_bits` as those are fixed in `SALT` array.